src/components/common/dialog.tsx

```diff
@@ -0,0 +1,1382 @@
+                                    }
+                                );
+                    this.#dialog._set_validation_promise(this.#path, val, prom);
+                }
+            );
+        }
+    }
+}
+
+interface DialogStateEvents {
+    changed(): void;
+}
+
+export class DialogState<V> extends EventEmitter<DialogStateEvents> {
+    values: V;
+
+    busy: boolean = false;
+    actions_disabled: boolean = false;
+    cancel_disabled: boolean = false;
+
+    error: unknown = null;
+
+    #validation_failed: boolean = false;
+    #online_validation: boolean = false;
+    #action_running: boolean = false;
+    #validation: Record<string, string | undefined> = { };
+    #validation_state: Record<string, DialogValidationState> = { };
+
+    /* eslint-disable no-use-before-define */
+    #validate_callback: (dlg: DialogState<V>) => void;
+    /* eslint-enable */
+
+    constructor(init: V, validate: (dlg: DialogState<V>) => void) {
+        super();
+        this.#validate_callback = validate;
+        this.values = init;
+    }
+
+    #update() {
+        this.busy = this.#action_running;
+        this.actions_disabled = this.#action_running || this.#validation_failed;
+        this.cancel_disabled = this.#action_running;
+        this.emit("changed");
+    }
+
+    #trigger_validation(): void {
+        this.#validation = { };
+        this.#validation_failed = false;
+        this.#validate_callback(this);
+        this.#update();
+    }
+
+    _probe_validation_cache(path: string, val: unknown): boolean {
+        if (path in this.#validation_state &&
+            Object.is(this.#validation_state[path].cached_value, val)) {
+            const state = this.#validation_state[path];
+            state.round_id = this.#validation;
+            const v = state.cached_result;
+            console.log("CACHE HIT", path, v);
+            this._set_validation(path, v);
+            return true;
+        } else
+            return false;
+    }
+
+    _get_validation_state(path: string): DialogValidationState | null {
+        if (path in this.#validation_state)
+            return this.#validation_state[path];
+        else
+            return null;
+    }
+
+    _set_validation_result(
+        path: string,
+        val: unknown,
+        result: string | undefined,
+    ) {
+        this.#validation_state[path] = {
+            cached_value: val,
+            cached_result: result,
+            timeout_id: 0,
+            promise: undefined,
+            round_id: this.#validation,
+        };
+        this._set_validation(path, result);
+    }
+
+    _set_validation_timeout(
+        path: string,
+        val: unknown,
+        delay: number,
+        func: () => void,
+    ) {
+        if (path in this.#validation_state) {
+            const state = this.#validation_state[path];
+            if (state.timeout_id) {
+                console.log("CANCEL TIMEOUT", path);
+                window.clearTimeout(state.timeout_id);
```

Comment:
These 5 added lines are not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-2449-e2dfc49b-20251217-161440-fedora-43-devel/Coverage/lcov/github-pr.diff.gcov.html)